Upgrading to 3.0.0 causes System.MissingMethodException: Method not found: 'Boolean Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers.CommandLineArgumentsHelper.TryGetIntArgFromDict

My service is built using Azure DevOps:

```
Current agent version: '2.175.2'
```

If I upgrade to version `3.0.0` my tests fail with error:

```
Starting: Running tests from ResourceProviderService.Tests.Functional
==============================================================================
Task         : .NET Core
Description  : Build, test, package, or publish a dotnet application, or run a custom dotnet command
Version      : 2.179.2
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/dotnet-core-cli
==============================================================================
C:\windows\system32\chcp.com 65001
Active code page: 65001
Info: .NET Core SDK/runtime 2.2 and 3.0 are now End of Life(EOL) and have been removed from all hosted agents. If you're using these SDK/runtimes on hosted agents, kindly upgrade to newer versions which are not EOL, or else use UseDotNet task to install the required version.
C:\dotnet\dotnet.exe test C:\azagent\_work\1\a\Service.Tests.Functional\Service.Tests.Functional.dll --logger trx --results-directory C:\azagent\_work\_temp --settings C:\azagent\_work\1\a/Service.Tests.Functional/RunSettings/FunctionalTests.Dev.RunSettings
Microsoft (R) Test Execution Command Line Tool Version 16.7.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

A total of 1 test files matched the specified pattern.
Testhost process exited with error: Unhandled exception. System.MissingMethodException: Method not found: 'Boolean Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers.CommandLineArgumentsHelper.TryGetIntArgFromDict(System.Collections.Generic.IDictionary`2<System.String,System.String>, System.String, Int32 ByRef)'.
   at Microsoft.VisualStudio.TestPlatform.TestHost.DefaultEngineInvoker.SetParentProcessExitCallback(IDictionary`2 argsDictionary)
   at Microsoft.VisualStudio.TestPlatform.TestHost.DefaultEngineInvoker.Invoke(IDictionary`2 argsDictionary)
   at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Run(String[] args)
   at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main(String[] args)
. Please check the diagnostic logs for more information.
Results File: C:\azagent\_work\_temp\testagent000000$_testagent000000_2021-01-12_19_59_37.trx
```

Rolling back to `1.3.0` makes the issue to disappear.

Other installed packages:

```xml
<ItemGroup>
  <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.3" />
  <PackageReference Include="MSTest.TestAdapter" Version="2.1.2" />
  <PackageReference Include="MSTest.TestFramework" Version="2.1.2" />
  <PackageReference Include="coverlet.collector" Version="1.3.0">
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
</ItemGroup>
```

Same issue for me, I was getting the following in Azure Devops with coverlet 3.0.2:

2021-02-10T22:37:32.8278999Z ##[error]Testhost process exited with error: Unhandled exception. System.MissingMethodException: Method not found: 'Boolean Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers.CommandLineArgumentsHelper.TryGetIntArgFromDict(System.Collections.Generic.IDictionary`2<System.String,System.String>, System.String, Int32 ByRef)'.
2021-02-10T22:37:32.8291747Z ##[error]   at Microsoft.VisualStudio.TestPlatform.TestHost.DefaultEngineInvoker.SetParentProcessExitCallback(IDictionary`2 argsDictionary)
2021-02-10T22:37:32.8293143Z ##[error]   at Microsoft.VisualStudio.TestPlatform.TestHost.DefaultEngineInvoker.Invoke(IDictionary`2 argsDictionary)
2021-02-10T22:37:32.8294283Z ##[error]   at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Run(String[] args)
2021-02-10T22:37:32.8295380Z ##[error]   at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main(String[] args)

Once I reverted back to 1.3.0 it worked fine again.

Let me know if you'd like any more details

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -8,6 +8,7 @@
     <PackageReference Update="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
     <PackageReference Update="Microsoft.Extensions.FileSystemGlobbing" Version="2.0.1" />
     <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.5.0" />
+    <!-- https://github.com/coverlet-coverage/coverlet/issues/1042 -->
     <PackageReference Update="Microsoft.TestPlatform.ObjectModel" Version="16.9.1" />
     <PackageReference Update="Mono.Cecil" Version="0.11.1" />
     <PackageReference Update="Moq" Version="4.10.1" />

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -8,7 +8,7 @@
     <PackageReference Update="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
     <PackageReference Update="Microsoft.Extensions.FileSystemGlobbing" Version="2.0.1" />
     <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.5.0" />
-    <PackageReference Update="Microsoft.TestPlatform.ObjectModel" Version="16.5.0" />
+    <PackageReference Update="Microsoft.TestPlatform.ObjectModel" Version="16.9.1" />
     <PackageReference Update="Mono.Cecil" Version="0.11.1" />
     <PackageReference Update="Moq" Version="4.10.1" />
     <!-- Do not upgrade this version or we won't support old SDK -->

